ibm/resource_ibm_private_dns_permitted_network.go

```diff
@@ -0,0 +1,154 @@
+			},
+		),
+
+		Schema: map[string]*schema.Schema{
+			pdnsInstanceID: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+
+			pdnsZoneID: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+
+			pdnsVpcCRN: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+		},
+	}
+}
+func resourceIBMPrivateDnsPermittedNetworkCreate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return err
+	}
+
+	instanceID := d.Get(pdnsInstanceID).(string)
+	zoneID := d.Get(pdnsZoneID).(string)
+	vpcCRN := d.Get(pdnsVpcCRN).(string)
+
+	createPermittedNetworkOptions := sess.NewCreatePermittedNetworkOptions(instanceID, zoneID)
+	permittedNetworkCrn, err := sess.NewPermittedNetworkVpc(vpcCRN)
+
+	createPermittedNetworkOptions.SetPermittedNetwork(permittedNetworkCrn)
+	createPermittedNetworkOptions.SetType(dnssvcsv1.CreatePermittedNetworkOptions_Type_Vpc)
+	response, _, err := sess.CreatePermittedNetwork(createPermittedNetworkOptions)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s/%s", instanceID, zoneID, *response.ID))
+	if err != nil {
+		log.Printf("[DEBUG]  err %s", isErrorToString(err))
+		return err
+	}
+
+	return resourceIBMPrivateDnsPermittedNetworkRead(d, meta)
+}
+
+func resourceIBMPrivateDnsPermittedNetworkRead(d *schema.ResourceData, meta interface{}) error {
+
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return err
+	}
+
+	id_set := strings.Split(d.Id(), "/")
+	getPermittedNetworkOptions := sess.NewGetPermittedNetworkOptions(id_set[0], id_set[1], id_set[2])
+	_, _, err = sess.GetPermittedNetwork(getPermittedNetworkOptions)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceIBMPrivateDnsPermittedNetworkUpdate(d *schema.ResourceData, meta interface{}) error {
+	return resourceIBMPrivateDnsPermittedNetworkRead(d, meta)
+}
+
+func resourceIBMPrivateDnsPermittedNetworkDelete(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return err
+	}
+
+	id_set := strings.Split(d.Id(), "/")
+	deletePermittedNetworkOptions := sess.NewDeletePermittedNetworkOptions(id_set[0], id_set[1], id_set[2])
+	_, _, reqErr := sess.DeletePermittedNetwork(deletePermittedNetworkOptions)
+
+	if reqErr != nil {
+		return reqErr
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func resourceIBMPrivateDnsPermittedNetworkExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return false, err
+	}
+
+	id_set := strings.Split(d.Id(), "/")
+	getPermittedNetworkOptions := sess.NewGetPermittedNetworkOptions(id_set[0], id_set[1], id_set[2])
+	_, _, err = sess.GetPermittedNetwork(getPermittedNetworkOptions)
+	if err != nil {
+		iserror, ok := err.(iserrors.RiaasError)
```

Comment:
remove iserrors.RiaasError just return same err which we get from SDK ..
https://github.com/IBM-Cloud/terraform-provider-ibm/blob/06df6ca39a6e811795cb4d7d4441125d7ac5add2/ibm/resource_ibm_api_gateway_endpoint_subscription.go#L258